### PR TITLE
ESYS: Fix session attribute automation problems

### PR DESCRIPTION
**Pull request by AndreasFuchsSIT**
Add fix and testcase.

Fixes: #1332

### CODE REVIEW

File: src/tss2-esys/esys_iutil.c [hunk 1]
```diff
@@ -611,13 +604,16 @@ iesys_restore_session_flags(ESYS_CONTEXT *esys_context)
...
     LOG_DEBUG("Restoring session attribs");
 
-    if (esys_context->enc_session) {
-        LOG_DEBUG("Orig Session Attrs 0x%x, altered Attrs x%x",
-                  esys_context->enc_session->origSessionAttributes,
-                  esys_context->enc_session->sessionAttributes);
+    for (int i = 0; i < 3; i++) {
+        RSRC_NODE_T *session = esys_context->session_tab[i];
+        IESYS_SESSION *rsrc_session = &session->rsrc.misc.rsrc_session;
+        if (session == NULL)
```
Comment (tstruk) [inline src/tss2-esys/esys_iutil.c]: Looks like we first dereference the session pointer and then check if it is NULL afterwards?

Comment (AndreasFuchsSIT) [reply src/tss2-esys/esys_iutil.c]: Totally true of course...
Monday then, or you just rewrite my patch...

Comment (tstruk) [reply src/tss2-esys/esys_iutil.c]: Monday is fine

Comment (AndreasFuchsSIT) [reply src/tss2-esys/esys_iutil.c]: Fixed

**Comment by AndreasFuchsSIT** [comment]
Should be all done and ready now...

**Comment by tstruk** [comment]
Just noticed that we have now two session flag test files:
test/integration/esys-session-attributes.int.c
test/integration/esys-auto-session-flags.int.c
Probably makes sense to merge them?

**Pull request merged**